openbas-api/src/main/java/io/openbas/rest/injector_contract/InjectorContractService.java

```diff
@@ -0,0 +1,127 @@
+    cq.orderBy(orders);
+
+    // Type Query
+    TypedQuery<Tuple> query = this.entityManager.createQuery(cq);
+
+    // -- Pagination --
+    query.setFirstResult((int) pageable.getOffset());
+    query.setMaxResults(pageable.getPageSize());
+
+    // -- EXECUTION --
+    List<InjectorContractOutput> injectorContractOutputs = execInjectorContract(query);
+
+    // -- Count Query --
+    Long total = countQuery(cb, this.entityManager, InjectorContract.class, specification);
+
+    return new PageImpl<>(injectorContractOutputs, pageable, total);
+  }
+
+
+  // -- CRITERIA BUILDER --
+
+  private void selectForInjectorContract(
+      @NotNull final CriteriaBuilder cb,
+      @NotNull final CriteriaQuery<Tuple> cq,
+      @NotNull final Root<InjectorContract> injectorContractRoot) {
+    // Joins
+    Join<InjectorContract, Payload> injectorContractPayloadJoin = createLeftJoin(injectorContractRoot, "payload");
+    Join<Payload, Collector> payloadCollectorJoin = injectorContractPayloadJoin.join("collector", JoinType.LEFT);
+    Join<InjectorContract, Injector> injectorContractInjectorJoin = createLeftJoin(injectorContractRoot, "injector");
+    // Array aggregations
+    Expression<String[]> attackPatternIdsExpression = createJoinArrayAggOnId(cb, injectorContractRoot, "attackPatterns");
+
+    // SELECT
+    cq.multiselect(
+        injectorContractRoot.get("id").alias("injector_contract_id"),
+        injectorContractRoot.get("labels").alias("injector_contract_labels"),
+        injectorContractRoot.get("content").alias("injector_contract_content"),
+        injectorContractRoot.get("platforms").alias("injector_contract_platforms"),
+        injectorContractPayloadJoin.get("type").alias("payload_type"),
+        payloadCollectorJoin.get("type").alias("collector_type"),
+        injectorContractInjectorJoin.get("type").alias("injector_contract_injector_type"),
+        attackPatternIdsExpression.alias("injector_contract_attack_patterns")
+    ).distinct(true);
+
+    // GROUP BY
+    cq.groupBy(Arrays.asList(
+        injectorContractRoot.get("id"),
+        injectorContractPayloadJoin.get("id"),
+        payloadCollectorJoin.get("id"),
+        injectorContractInjectorJoin.get("id")
+    ));
+  }
+
+
+  private List<InjectorContractOutput> execInjectorContract(TypedQuery<Tuple> query) {
+    return query.getResultList()
+        .stream()
+        .map(tuple -> new InjectorContractOutput(
+            tuple.get("injector_contract_id", String.class),
+            tuple.get("injector_contract_labels", Map.class),
+            tuple.get("injector_contract_content", String.class),
+            tuple.get("injector_contract_platforms", String[].class),
+            tuple.get("payload_type", String.class),
+            tuple.get("collector_type", String.class),
+            tuple.get("injector_contract_injector_type", String.class),
+            tuple.get("injector_contract_attack_patterns", String[].class)
```

Comment:
We need to keep in mind that the XLS import will need another property here

Reply:
i don't understand the impact

Reply:
That was just a reminder that we will need the add the injector_contract_import_available here for the XLS import PR. Nothing to worry about in this PR :)